pkg/controller/controller_instance.go

```diff
@@ -1474,6 +1483,13 @@ func (c *controller) processProvisionFailure(instance *v1beta1.ServiceInstance,
 		err = fmt.Errorf(failedCond.Message)
 	} else {
 		clearServiceInstanceCurrentOperation(instance)
+
+		if instance.DeletionTimestamp != nil {
```

Comment:
could you extract this code to a separate method instead of copy-pasting? it will make a cleanup in a follow-up PR much easier.